[Flaky Test]: TestOTelManager_Run – otel collector never got healthy

### Failing test case

TestOTelManager_Run

### Error message

otel collector never got healthy

### Build

https://buildkite.com/elastic/elastic-agent/builds/14279

### OS

Linux

### Stacktrace and notes

```markdown
=== RUN   TestOTelManager_Run
    manager_test.go:123: 
        	Error Trace:	/opt/buildkite-agent/builds/bk-agent-prod-gcp-1732147373514009164/elastic/elastic-agent/internal/pkg/otel/manager/manager_test.go:123
        	            				/opt/buildkite-agent/builds/bk-agent-prod-gcp-1732147373514009164/elastic/elastic-agent/internal/pkg/otel/manager/manager_test.go:169
        	Error:      	Condition never satisfied
        	Test:       	TestOTelManager_Run
        	Messages:   	otel collector never got healthy
    manager_test.go:146: otel collector never got healthy: <nil> (latest err: <nil>)
```

This happened again in https://buildkite.com/elastic/elastic-agent/builds/17563#0195426f-9bc9-43ce-8f7e-94880f4adb80

```
=== RUN   TestOTelManager_Run
    manager_test.go:121: 
        	Error Trace:	C:/buildkite-agent/builds/bk-agent-prod-gcp-1740576392206160015/elastic/elastic-agent/internal/pkg/otel/manager/manager_test.go:121
        	            				C:/buildkite-agent/builds/bk-agent-prod-gcp-1740576392206160015/elastic/elastic-agent/internal/pkg/otel/manager/manager_test.go:171
        	Error:      	Condition never satisfied
        	Test:       	TestOTelManager_Run
        	Messages:   	otel collector never got healthy
    manager_test.go:144: otel collector never got healthy: &{Event:0xc000972f90 ComponentStatusMap:map[pipeline:logs:0xc000cfcba0 pipeline:metrics:0xc000cfcc60 pipeline:traces:0xc000cfcc00]} (latest err: <nil>)
--- FAIL: TestOTelManager_Run (301.05s)
```

Looks like we should improve the debug information in this test first, as the status map doesn't tell us anything useful the way it is right now.

--- a/internal/pkg/otel/manager/manager_test.go
+++ b/internal/pkg/otel/manager/manager_test.go
@@ -13,6 +13,8 @@ import (
 	"testing"
 	"time"
 
+	"gopkg.in/yaml.v2"
+
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
@@ -127,7 +129,7 @@ func TestOTelManager_Run(t *testing.T) {
 				return true
 			}
 			latest := getLatestStatus()
-			if latest == nil || latest.Event.Status() != componentstatus.StatusOK {
+			if latest == nil || latest.Status() != componentstatus.StatusOK {
 				return false
 			}
 			return true
@@ -143,7 +145,7 @@ func TestOTelManager_Run(t *testing.T) {
 			if !errors.Is(runErr, context.Canceled) {
 				t.Fatalf("otel manager never got healthy and the otel manager returned unexpected error: %v (latest status: %+v) (latest err: %v)", runErr, lastStatus, lastErr)
 			}
-			t.Fatalf("otel collector never got healthy: %+v (latest err: %v)", lastStatus, lastErr)
+			t.Fatalf("otel collector never got healthy: %s (latest err: %v)", statusToYaml(lastStatus), lastErr)
 		}
 		latestErr := getLatestErr()
 		require.NoError(t, latestErr, "runtime errored")
@@ -236,4 +238,37 @@ outer:
 		}
 	}
 	assert.Error(t, err, "otel manager should have returned an error")
+}
+
+func statusToYaml(s *status.AggregateStatus) string {
+	printable := toSerializableStatus(s)
+	yamlBytes, _ := yaml.Marshal(printable)
+	return string(yamlBytes)
+}
+
+type serializableStatus struct {
+	Status             string
+	Error              error
+	Timestamp          time.Time
+	ComponentStatusMap map[string]serializableStatus
+}
+
+// converts the status.AggregateStatus to a serializable form. The normal status is structured in a way where
+// serialization based on reflection doesn't give the right result.
+func toSerializableStatus(s *status.AggregateStatus) *serializableStatus {
+	if s == nil {
+		return nil
+	}
+
+	outputComponentStatusMap := make(map[string]serializableStatus, len(s.ComponentStatusMap))
+	for k, v := range s.ComponentStatusMap {
+		outputComponentStatusMap[k] = *toSerializableStatus(v)
+	}
+	outputStruct := &serializableStatus{
+		Status:             s.Status().String(),
+		Error:              s.Err(),
+		Timestamp:          s.Timestamp(),
+		ComponentStatusMap: outputComponentStatusMap,
+	}
+	return outputStruct
 }

